src/main/java/com/mathworks/ci/BuildArtifactAction.java

```diff
@@ -0,0 +1,208 @@
+package com.mathworks.ci;
+
+import hudson.FilePath;
+import hudson.model.Action;
+import hudson.model.Run;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.annotation.CheckForNull;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+
+public class BuildArtifactAction implements Action {
+    private Run<?, ?> build;
```

Comment:
the build could be part of Freestyle aas well as folder type or any other type. We suport our build for all types of build along with Freestyle this will be known only when user creates the job of particular type.

Reply:
I know this is marked as resolved, but I'm having some trouble understanding. This should only support our Run Build integration from freestyle or Jenkinsfile right? I suppose these are different types, but do they derive from a common type? That could be preferable than using unknown.

Reply:
Specific type here could be freestyle and or pipeline project etc however Unknown types gives wider range to support this action. if we have to make it for specific type then in the perticular builder class we need to change it for that perticular type . In that case these build steps will be avaialble only when those Item types (Freestyle/ Pipeline) are selected.